supervision/dataset/classification.py

```diff
@@ -0,0 +1,239 @@
+
+        Args:
+            root_directory_path (str): The path to the root directory of the images with which you are working.
+            output_directory_path (str): The path to the directory where the dataset will be saved.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset(...)
+
+            >>> cd.as_multiclass_folder_structure(
+            ...     root_directory_path="./images",
+            ...     output_directory_path="./out",
+            ... )
+        """
+        if not os.path.exists(output_directory_path):
+            os.makedirs(output_directory_path)
+
+        for split in ["train", "test", "valid"]:
+            if not os.path.exists(output_directory_path + "/" + split):
+                os.makedirs(output_directory_path + "/" + split)
+
+        train, test = self.split(split_ratio=0.8)
+        train, val = train.split(split_ratio=0.8)
+
+        for split, dataset in zip(["train", "test", "valid"], [train, test, val]):
+            for class_name in self.classes:
+                if not os.path.exists(
+                    output_directory_path + "/" + split + "/" + class_name
+                ):
+                    os.makedirs(output_directory_path + "/" + split + "/" + class_name)
+
+                for image in dataset.images:
+                    if dataset.annotations[image].class_id[0] == self.classes.index(
+                        class_name
+                    ):
+                        full_dir = os.path.join(root_directory_path, image)
+
+                        cv2.imwrite(
+                            output_directory_path
+                            + "/"
+                            + split
+                            + "/"
+                            + class_name
+                            + "/"
+                            + image,
+                            cv2.imread(full_dir),
+                        )
+
+    @classmethod
+    def from_multiclass_folder_structure(
+        cls, root_directory_path: str
+    ) -> "ClassificationDataset":
+        """
+        Load data from a multiclass folder structure into a ClassificationDataset.
+
+        Args:
+            root_directory_path (str): The path to the dataset directory.
+
+        Returns:
+            ClassificationDataset: The dataset.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset.from_multiclass_folder_structure(
+            ...     root_directory_path="./dataset",
+            ... )
+        """
+        classes = os.listdir(root_directory_path + "/train")
+        classes.sort()
+
+        images = {}
+        annotations = {}
+
+        for split in ["train", "test", "val"]:
+            for class_name in classes:
+                for image in os.listdir(
+                    root_directory_path + "/" + split + "/" + class_name
```

Comment:
Please do not build paths using `+ "/" +` . This is not secure. Python has the tools to do that. For example, `os.path.join`.

Reply:
```
path = "/home" + "/" + "user" + "/" + "documents" # This will work on Unix-based systems, but not on Windows
```
```
path = "C:\\Users" + "\\" + "user" + "\\" + "documents" # This is how paths are typically structured on Windows
```